File: javascript/src/model/LocationInfo.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.RingCentralApi) {
      root.RingCentralApi = {};
    }
    root.RingCentralApi.LocationInfo = factory(root.RingCentralApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The LocationInfo model module.
   * @module model/LocationInfo
   * @version v1.0
   */

  /**
   * Constructs a new <code>LocationInfo</code>.
   * @alias module:model/LocationInfo
   * @class
   */
  var exports = function() {







  };

  /**
   * Constructs a <code>LocationInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/LocationInfo} obj Optional instance to populate.
   * @return {module:model/LocationInfo} The populated <code>LocationInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('uri')) {
        obj['uri'] = ApiClient.convertToType(data['uri'], 'String');
      }
      if (data.hasOwnProperty('areaCode')) {
        obj['areaCode'] = ApiClient.convertToType(data['areaCode'], 'String');
      }
      if (data.hasOwnProperty('city')) {
        obj['city'] = ApiClient.convertToType(data['city'], 'String');
      }
      if (data.hasOwnProperty('npa')) {
        obj['npa'] = ApiClient.convertToType(data['npa'], 'String');
      }
      if (data.hasOwnProperty('nxx')) {
        obj['nxx'] = ApiClient.convertToType(data['nxx'], 'String');
      }
      if (data.hasOwnProperty('state')) {
        obj['state'] = ApiClient.convertToType(data['state'], 'String');
      }
    }
    return obj;
  }


  /**
   * Canonical URI of a location
   * @member {String} uri
   */
  exports.prototype['uri'] = undefined;

  /**
   * Area code of the location
   * @member {String} areaCode
   */
  exports.prototype['areaCode'] = undefined;

  /**
   * Official name of the city, belonging to the certain state
   * @member {String} city
   */
  exports.prototype['city'] = undefined;

  /**
   * Area code of the location (3-digit usually), according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan
   * @member {String} npa
   */
  exports.prototype['npa'] = undefined;

  /**
   * Central office code of the location, according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan
   * @member {String} nxx
   */
  exports.prototype['nxx'] = undefined;

  /**
   * ID and URI of the state this location belongs to, see State Info
   * @member {String} state
   */
  exports.prototype['state'] = undefined;




  return exports;
}));

File: csharp/src/main/csharp/IO/Swagger/Model/LocationInfo.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace IO.Swagger.Model
{
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class LocationInfo :  IEquatable<LocationInfo>
    { 
    
        /// <summary>
        /// Initializes a new instance of the <see cref="LocationInfo" /> class.
        /// Initializes a new instance of the <see cref="LocationInfo" />class.
        /// </summary>
        /// <param name="Uri">Canonical URI of a location.</param>
        /// <param name="AreaCode">Area code of the location.</param>
        /// <param name="City">Official name of the city, belonging to the certain state.</param>
        /// <param name="Npa">Area code of the location (3-digit usually), according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan.</param>
        /// <param name="Nxx">Central office code of the location, according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan.</param>
        /// <param name="State">ID and URI of the state this location belongs to, see State Info.</param>

        public LocationInfo(string Uri = null, string AreaCode = null, string City = null, string Npa = null, string Nxx = null, string State = null)
        {
            this.Uri = Uri;
            this.AreaCode = AreaCode;
            this.City = City;
            this.Npa = Npa;
            this.Nxx = Nxx;
            this.State = State;
            
        }
        
    
        /// <summary>
        /// Canonical URI of a location
        /// </summary>
        /// <value>Canonical URI of a location</value>
        [DataMember(Name="uri", EmitDefaultValue=false)]
        public string Uri { get; set; }
    
        /// <summary>
        /// Area code of the location
        /// </summary>
        /// <value>Area code of the location</value>
        [DataMember(Name="areaCode", EmitDefaultValue=false)]
        public string AreaCode { get; set; }
    
        /// <summary>
        /// Official name of the city, belonging to the certain state
        /// </summary>
        /// <value>Official name of the city, belonging to the certain state</value>
        [DataMember(Name="city", EmitDefaultValue=false)]
        public string City { get; set; }
    
        /// <summary>
        /// Area code of the location (3-digit usually), according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan
        /// </summary>
        /// <value>Area code of the location (3-digit usually), according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan</value>
        [DataMember(Name="npa", EmitDefaultValue=false)]
        public string Npa { get; set; }
    
        /// <summary>
        /// Central office code of the location, according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan
        /// </summary>
        /// <value>Central office code of the location, according to the NANP number format, that can be summarized as NPA-NXX-xxxx and covers Canada, the United States, parts of the Caribbean Sea, and some Atlantic and Pacific islands. See for details North American Numbering Plan</value>
        [DataMember(Name="nxx", EmitDefaultValue=false)]
        public string Nxx { get; set; }
    
        /// <summary>
        /// ID and URI of the state this location belongs to, see State Info
        /// </summary>
        /// <value>ID and URI of the state this location belongs to, see State Info</value>
        [DataMember(Name="state", EmitDefaultValue=false)]
        public string State { get; set; }
    
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class LocationInfo {\n");
            sb.Append("  Uri: ").Append(Uri).Append("\n");
            sb.Append("  AreaCode: ").Append(AreaCode).Append("\n");
            sb.Append("  City: ").Append(City).Append("\n");
            sb.Append("  Npa: ").Append(Npa).Append("\n");
            sb.Append("  Nxx: ").Append(Nxx).Append("\n");
            sb.Append("  State: ").Append(State).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as LocationInfo);
        }

        /// <summary>
        /// Returns true if LocationInfo instances are equal
        /// </summary>
        /// <param name="other">Instance of LocationInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(LocationInfo other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.Uri == other.Uri ||
                    this.Uri != null &&
                    this.Uri.Equals(other.Uri)
                ) && 
                (
                    this.AreaCode == other.AreaCode ||
                    this.AreaCode != null &&
                    this.AreaCode.Equals(other.AreaCode)
                ) && 
                (
                    this.City == other.City ||
                    this.City != null &&
                    this.City.Equals(other.City)
                ) && 
                (
                    this.Npa == other.Npa ||
                    this.Npa != null &&
                    this.Npa.Equals(other.Npa)
                ) && 
                (
                    this.Nxx == other.Nxx ||
                    this.Nxx != null &&
                    this.Nxx.Equals(other.Nxx)
                ) && 
                (
                    this.State == other.State ||
                    this.State != null &&
                    this.State.Equals(other.State)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.Uri != null)
                    hash = hash * 59 + this.Uri.GetHashCode();
                
                if (this.AreaCode != null)
                    hash = hash * 59 + this.AreaCode.GetHashCode();
                
                if (this.City != null)
                    hash = hash * 59 + this.City.GetHashCode();
                
                if (this.Npa != null)
                    hash = hash * 59 + this.Npa.GetHashCode();
                
                if (this.Nxx != null)
                    hash = hash * 59 + this.Nxx.GetHashCode();
                
                if (this.State != null)
                    hash = hash * 59 + this.State.GetHashCode();
                
                return hash;
            }
        }

    }
}
